### PR TITLE
overlord: make changes conflict with remodel

### PR DESCRIPTION
**Pull request by pedronis**
This makes starting snap operations behave similarly to before seeding is done when a remodeling is in progress. It also makes any new change conflict with a remodel change.

It introduces devicestate.Remodeling.

Also adds a TODO about whether remodelings/complex remodelings should wait for other changes to be done before proceeding?

### CODE REVIEW

**Comment by pedronis** [comment]
based on #6855 now

File: overlord/devicestate/devicestate.go [hunk 1]
```diff
@@ -535,3 +541,16 @@ func Remodel(st *state.State, new *asserts.Model) (*state.Change, error) {
...
+
+// Remodeling returns true whether there's a remodeling in progress
+func Remodeling(st *state.State) bool {
+	for _, chg := range st.Changes() {
+		if chg.IsReady() {
+			continue
+		}
+		if chg.Kind() == "remodel" {
+			return true
+		}
```
Comment (stolowski) [inline overlord/devicestate/devicestate.go]: Nitpick, how about slightly more compact 
```
if !chg.IsReady() && kind == "remodel" {
   return true
}```

File: overlord/devicestate/devicestate.go [hunk 2]
```diff
@@ -491,6 +492,11 @@ func Remodel(st *state.State, new *asserts.Model) (*state.Change, error) {
 		return nil, fmt.Errorf("cannot remodel to different gadgets yet")
 	}
 
+	// TODO: should we run a remodel only while no other change is
+	// running?  do we add a task upfron that waits for that to be
+	// true? Do we do this only for the more complicated cases
+	// (anything more than adding required-snaps really)?
```
Comment (stolowski) [inline overlord/devicestate/devicestate.go]: Without spending too much time thinking about bigger picture of this question, I'd say that - given the specific uses of this feature and the fact that is generally going to be used once (is it a sensible assumption?), I'd strive to keep it simple (e.g. remodel only when no other changes are in flight) and not try to optimize, at least for the initial version.

File: overlord/snapstate/snapstatetest/devicectx.go [hunk 3]
```diff
@@ -62,8 +62,16 @@ func MockDeviceContext(deviceCtx snapstate.DeviceContext) (restore func()) {
 		}
 		return deviceCtx, nil
 	}
-	return ReplaceDeviceCtxHook(deviceCtxHook)
-
+	r1 := ReplaceDeviceCtxHook(deviceCtxHook)
+	// for convenience reflect from the context whether there is a
+	// remodeling
+	r2 := ReplaceRemodelingHook(func(*state.State) bool {
```
Comment (stolowski) [inline overlord/snapstate/snapstatetest/devicectx.go]: Nitpick^2: s/r1/restoreDeviceCtxHook/ and s/r2/restoreRemodelingHook/

**Pull request merged**